diff --git a/src/sagemaker/image_uris.py b/src/sagemaker/image_uris.py
--- a/src/sagemaker/image_uris.py
+++ b/src/sagemaker/image_uris.py
@@ -1,5 +1,5 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0
+# SPDX-License-Identifier: Apache-2.0
 
 """Placeholder docstring"""
 from __future__ import absolute_import
@@ -8,6 +8,7 @@
 import re
 from typing import Optional, Dict, List
 
+from sagemaker import Session
 
 logger = logging.getLogger(__name__)
 
@@ -16,6 +17,7 @@
     "inference",
     "training",
     "eia",
+    "inference-graviton",
     "inference-neuron",
     "inference-neuronx",
     "training-neuronx",
@@ -23,6 +25,7 @@
 ]
 
 _IMAGE_SCOPE_TO_FRAMEWORK_OVERRIDE = {
+    "inference-graviton": "inference",
     "inference-neuron": "inference",
     "inference-neuronx": "inference",
     "training-neuronx": "training",
@@ -30,6 +33,7 @@
 }
 
 _SCOPE_TO_PROCESSOR_MAP = {
+    "inference-graviton": "cpu",
     "inference-neuron": "inf",
     "inference-neuronx": "trn",
     "training-neuronx": "trn",
@@ -37,6 +41,7 @@
 }
 
 _SCOPE_TO_PROCESSOR_SUFFIX_MAP = {
+    "inference-graviton": "",
     "inference-neuron": "-neuron",
     "inference-neuronx": "-neuronx",
     "training-neuronx": "-neuronx",
@@ -44,6 +49,7 @@
 }
 
 _SCOPE_TO_JOB_TYPE_MAP = {
+    "inference-graviton": "inference",
     "inference-neuron": "inference",
     "inference-neuronx": "inference",
     "training-neuronx": "training",
@@ -51,6 +57,7 @@
 }
 
 _SCOPE_TO_FRAMEWORK_MAP = {
+    "inference-graviton": None,
     "inference-neuron": "neuron",
     "inference-neuronx": "neuronx",
     "training-neuronx": "neuronx",
@@ -58,6 +65,7 @@
 }
 
 _SCOPE_TO_FRAMEWORK_VERSION_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -65,6 +73,7 @@
 }
 
 _SCOPE_TO_PY_VERSION_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -72,6 +81,7 @@
 }
 
 _SCOPE_TO_ACCELERATOR_TYPE_MAP = {
+    "inference-graviton": None,
     "inference-neuron": "inf",
     "inference-neuronx": "trn",
     "training-neuronx": "trn",
@@ -79,6 +89,7 @@
 }
 
 _SCOPE_TO_DISTRIBUTION_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -86,6 +97,7 @@
 }
 
 _SCOPE_TO_BASE_FRAMEWORK_VERSION_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -93,6 +105,7 @@
 }
 
 _SCOPE_TO_TRAINING_COMPILER_CONFIG_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -100,6 +113,7 @@
 }
 
 _SCOPE_TO_CONTAINER_VERSION_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -107,6 +121,7 @@
 }
 
 _SCOPE_TO_IMAGE_URI_PATTERN_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -114,6 +129,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REGISTRIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -121,6 +137,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REPOSITORIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -128,6 +145,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -135,6 +153,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PY_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -142,6 +161,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PROCESSORS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -149,6 +169,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_TAG_PREFIXES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -156,6 +177,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_IMAGE_SCOPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -163,6 +185,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_JOB_TYPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -170,6 +193,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_FRAMEWORKS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -177,6 +201,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_FRAMEWORK_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -184,6 +209,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PY_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -191,6 +217,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_ACCELERATOR_TYPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -198,6 +225,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_DISTRIBUTIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -205,6 +233,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_BASE_FRAMEWORK_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -212,6 +241,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_TRAINING_COMPILER_CONFIGS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -219,6 +249,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_CONTAINER_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -226,6 +257,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_IMAGE_URI_PATTERNS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -233,6 +265,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REGISTRIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -240,6 +273,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REPOSITORIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -247,6 +281,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -254,6 +289,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PY_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -261,6 +297,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PROCESSORS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -268,6 +305,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_TAG_PREFIXES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -275,6 +313,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_IMAGE_SCOPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -282,6 +321,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_JOB_TYPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -289,6 +329,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_FRAMEWORKS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -296,6 +337,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_FRAMEWORK_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -303,6 +345,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PY_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -310,6 +353,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_ACCELERATOR_TYPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -317,6 +361,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_DISTRIBUTIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -324,6 +369,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_BASE_FRAMEWORK_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -331,6 +377,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_TRAINING_COMPILER_CONFIGS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -338,6 +385,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_CONTAINER_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -345,6 +393,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_IMAGE_URI_PATTERNS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -352,6 +401,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REGISTRIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -359,6 +409,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_REPOSITORIES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -366,6 +417,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -373,6 +425,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PY_VERSIONS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -380,6 +433,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_PROCESSORS_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -387,6 +441,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_TAG_PREFIXES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -394,6 +449,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_IMAGE_SCOPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -401,6 +457,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_JOB_TYPES_MAP = {
+    "inference-graviton": None,
     "inference-neuron": None,
     "inference-neuronx": None,
     "training-neuronx": None,
@@ -408,6 +465,7 @@
 }
 
 _SCOPE_TO_OVERRIDE_FRAMEWORKS_MAP